protocols/autonat/tests/test_server.rs

```diff
@@ -252,8 +258,11 @@ async fn test_dial_error() {
                     break;
                 }
                 SwarmEvent::Dialing(peer) => assert_eq!(peer, client_id),
-                SwarmEvent::NewListenAddr { .. } => {}
-                _ => panic!("Unexpected Swarm Event"),
```

Comment:
``` diff
-                 _ => panic!("Unexpected Swarm Event"),
+                 e => panic!("Unexpected Swarm Event: {:?}", e),
```

I would as well be fine to only log the error for now, thus enabling us to debug the actual issue.

Reply:
I am bit undecided on this. But I think panicking by "blanked-implementation" could cause headaches in the future. If for example a new swarm event is added that can happen here, the test could suddenly fail. For someone not familiar with this protocol it could take a while to figure out if the error is an actual error in the implementation, or just in the test itself because it is not up-to-date anymore. Even worse if it's a flaky error, like the one that caused #2440 to fail.

On the other hand I generally think it's good to test as much as possible, so that we also detect edge cases and unexpected interference. Considering that most swarm events don't just "randomly" occur, I am mostly concerned only about listener events. E.g. with a TCP transport a listener event is reported each time a network interface becomes available/ unavailable, which can happen at any time. 
So we could find a middle way by ignoring all listener events, and panicking on all other events like you proposed. Wdyt?

Reply:
> a TCP transport a listener event is reported each time a network interface becomes available/ unavailable, which can happen at any time. So we could find a middle way by ignoring all listener events, and panicking on all other events like you proposed. Wdyt?

The test setup should be deterministic enough for this to not happen though, shouldn't it?

Reply:
I am not too familiar with the tcp transport, so please correct me if I am wrong. 
But in some of these tests we spawn a TCP Listener, and from what I understand it reports an event each time the underlying `IfWatcher` reports an network interface change. And since that depends on the OS and network, I would think it is not deterministic?

Either way, I'll just change it to use the `MemoryTransport` and follow the above suggestion from @mxinden to panic in all cases and print the unexpected event.

Reply:
Correction: The client has to listen on a TCP address. For DOS attack prevention the server only dials addresses that are bases on the observed sender IP, an refuses the dial if there is none. See https://github.com/libp2p/specs/pull/369. 
So we have to stick with the (nondeterministic) TCP transport in the tests.